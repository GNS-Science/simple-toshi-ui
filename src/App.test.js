import TestRenderer from 'react-test-renderer';
import {createMockEnvironment, MockPayloadGenerator} from "relay-test-utils";
import AppStrongMotionStationQuery from './__generated__/AppStrongMotionStationQuery.graphql';
import AppRoot from './App';

/*
 * 
 * https://relay.dev/docs/guides/testing-relay-components/
 *
 * https://medium.com/dooboolab/testing-relay-hook-with-react-testing-library-d432f903b8d2
 * 
*/

describe('AppRoot component', () => {

it('displays `Loading...` while waiting for data', () => {
	const environment = createMockEnvironment();
  	const testRenderer = TestRenderer.create(
        <AppRoot environment={environment} />
     );
  	//expect(testRenderer.toJSON()).toContain('Loading OK...')
  	expect(testRenderer).toMatchSnapshot() //contains ProgressCircle
})

it('displays a StrongMotionStaton using mock graphql payload', () => {

	// //lets us mock out the graphql service (amongst other things)
	const environment = createMockEnvironment();

	//this defines the data we want our query to return
	const mockResolver = {
			StrongMotionStaton: (): any => ({
				id: 'id',
				site_code: 'site_code',
				site_class: 'A',
				created: '2020-10-10T23:00:00+00:00',
				soft_clay_or_peat: false,
				Vs30_mean: [23]
			}),
	};
 
 	//create a query operation
	environment.mock.queuePendingOperation(AppStrongMotionStationQuery, {A:1})

	//queu up the response we want returned
	environment.mock.queueOperationResolver(
  		(operation) => MockPayloadGenerator.generate(operation, mockResolver),
	);
	
	//create the test subject
  	const testRenderer = TestRenderer.create(
        <AppRoot environment={environment} />
     );


  	/*
  	 * this may be useful later ....
  	 */
	  // 	TestRenderer.act(() => {
	  // 		jest.useFakeTimers();
	  // 		// console.log("all ops:", environment.mock.getAllOperations());
	  //        // trigger the loading - click a button, emit an event, etc. or .
	  // 		jest.runAllImmediates();
	  //        console.log(environment.mock.getMostRecentOperation());
	  //        console.log(testRenderer.toJSON());
      // 	});

  	// assertions
    expect(testRenderer).toBeTruthy()
    expect(testRenderer).toMatchSnapshot() //will fail if our response is changed
  })
})