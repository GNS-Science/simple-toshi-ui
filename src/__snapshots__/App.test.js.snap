// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoot component displays a StrongMotionStaton using mock graphql payload 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      This data arrives via graphql query from a dev server (see toshi-api flask-relay)
    </p>
    <p />
    <p>
      <strong>
        StrongMotionStation
      </strong>
       id: "U3Ryb25nTW90aW9uU3RhdGlvbjow"
    </p>
    <p>
      site_code: 
      &lt;mock-value-for-field-"site_code"&gt;
    </p>
    <p>
      created: 
      &lt;mock-value-for-field-"created"&gt;
    </p>
  </header>
</div>
`;
